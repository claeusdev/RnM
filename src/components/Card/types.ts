import { Character } from "types";

export interface CharacterProps {
    character: Character;
  }