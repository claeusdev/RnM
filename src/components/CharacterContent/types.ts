import { Character, Location } from "types";


export interface CharacterContentProps {
  gender: string;
  location: Location;
  origin: { name: string };
  species: string;
}
