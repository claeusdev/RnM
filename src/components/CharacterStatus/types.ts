export interface CharacterStatusProps {
  status: string;
}
