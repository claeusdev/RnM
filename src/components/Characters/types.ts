import { Character } from "types";

export type CharactersProps = {
    page: string
    results: Character[]
}