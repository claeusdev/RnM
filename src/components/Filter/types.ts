export interface FilterProps {
    updatePageNumber: (pageNumber: number) => void;
}
