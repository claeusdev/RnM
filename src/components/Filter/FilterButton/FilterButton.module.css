.filterBtn {
    position: relative;
    display: block;
    min-width: calc(1.4285714286em + 0.875rem + 2px);
    margin: 0;
    padding: 0.4375rem 0.75rem;
    font-size: .875rem;
    line-height: 1.4285714286;
    color: #626976;
    background: #fff;
    text-align: center;
    cursor: pointer;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 1px solid #d9dbde;
    border-radius: 3px;
    transition: border-color .3s,background .3s,color .3s;
  }