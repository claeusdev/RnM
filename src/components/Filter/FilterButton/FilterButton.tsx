import styles from './FilterButton.module.css';

interface FilterButtonProps {
  input: string;
  task: (input: string) => void;
  updatePageNumber: (pageNumber: number) => void;
  index: number;
  name: string;
}

const FilterButton: React.FC<FilterButtonProps> = ({
  input,
  task,
  updatePageNumber,
  index,
  name,
}) => {
  return (
    <>
      <style>
        {`
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        `}
      </style>

      <div className="form-check">
        <input
          className="form-check-input x"
          type="radio"
          name={name}
          id={`${name}-${index}`}
        />
        <label
          onClick={(x) => {
            task(input);
            updatePageNumber(1);
          }}
          className={styles.filterBtn}
          htmlFor={`${name}-${index}`}
        >
          {input}
        </label>
      </div>
    </>
  );
};

export default FilterButton;
