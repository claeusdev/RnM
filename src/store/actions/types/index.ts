export enum ActionTypes {
    LOAD_CHARACTERS = "LOAD_CHARACTERS",
    LOAD_CHARACTERS_SUCCESS = "LOAD_CHARACTERS_SUCCESS",
    LOAD_CHARACTERS_ERROR = "LOAD_CHARACTERS_ERROR",
    FILTER_CHARACTERS = "FILTER_CHARACTERS",
    FILTER_CHARACTERS_SUCCESS = "FILTER_CHARACTERS_SUCCESS",
    FILTER_CHARACTERS_ERROR = "FILTER_CHARACTERS_ERROR",
    LOAD_CHARACTER = "LOAD_CHARACTER",
    LOAD_CHARACTER_SUCCESS = "LOAD_CHARACTER_SUCCESS",
    LOAD_CHARACTER_ERROR = "LOAD_CHARACTER_ERROR",
    SET_EPISODES = "SET_EPISODES"
}