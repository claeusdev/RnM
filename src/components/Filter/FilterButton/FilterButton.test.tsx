import { render } from "@testing-library/react";
import { testCharacter } from "types/testData";
import FilterButton from "./FilterButton";

test("renders filter button with all props", () => {
  const component = render(<FilterButton input="" task={() => jest.fn()} updatePageNumber={jest.fn()} index={1} name={testCharacter.name}/>);
  expect(component).toMatchSnapshot();
});
