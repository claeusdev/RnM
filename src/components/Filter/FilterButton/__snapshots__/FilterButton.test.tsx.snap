// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders filter button with all props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style>
        
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
      </style>
      <div
        class="form-check"
      >
        <input
          class="form-check-input x"
          id="Rick Sanchez-1"
          name="Rick Sanchez"
          type="radio"
        />
        <label
          class="filterBtn"
          for="Rick Sanchez-1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <style>
      
          .x:checked + label {
            background-color: #0b5ed7;
            color: white;
          }
          input[type="radio"] {
            display: none;
          }
        
    </style>
    <div
      class="form-check"
    >
      <input
        class="form-check-input x"
        id="Rick Sanchez-1"
        name="Rick Sanchez"
        type="radio"
      />
      <label
        class="filterBtn"
        for="Rick Sanchez-1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
